code/_core/obj/item/clothing/belt/bandolier.dm

```diff
@@ -0,0 +1,288 @@
+/obj/item/clothing/belt/bandolier
+	name = "bandolier"
+	desc = "For the aspiring sweeper."
+	desc_extended = "A not-so-fancy bandolier meant to hold a number of shotgun shells. ALT+Click to grab additional shells while already holding a shell."
+	rarity = RARITY_UNCOMMON
+	icon = 'icons/obj/item/clothing/belts/bandolier_12.dmi'
+	worn_layer = LAYER_MOB_CLOTHING_COAT
+
+	defense_rating = list(
+		BLADE = AP_AXE,
+		BLUNT = AP_AXE,
+		PIERCE = AP_SWORD
+	)
+
+	rarity = RARITY_UNCOMMON
+	value = 200
+
+	var/list/stored_shells = list()
+	var/shell_count = 0
+
+	var/max_shells = 0
+
+/obj/item/clothing/belt/bandolier/save_item_data(var/save_inventory = TRUE)
+	. = ..()
+	.["stored_shells"] = list()
+	for(var/k in stored_shells)
+		var/v = stored_shells[k]
+		.["stored_shells"][k] = v
+
+/obj/item/clothing/belt/bandolier/load_item_data_pre(var/mob/living/advanced/player/P,var/list/object_data)
+	. = ..()
+	for(var/k in object_data["stored_shells"])
+		var/v = object_data["stored_shells"][k]
+		stored_shells[text2path(k)] = v
+
+/obj/item/clothing/belt/bandolier/Finalize()
+	. = ..()
+	update_sprite()
+
+/obj/item/clothing/belt/bandolier/proc/get_shell_count()
```

Comment:
Use length(stored_shells) for this. it does the same thing here but faster and much better.

Reply:
Yep, did it. Any particular reason that's used in the quiver dm then?